lib/spack/spack/url.py

```diff
@@ -188,6 +188,10 @@ def parse_version_offset(path, debug=False):
         # e.g. https://github.com/petdance/ack/tarball/1.93_02
         (r'github.com/.+/(?:zip|tar)ball/v?((\d+\.)+\d+_(\d+))$', path),
 
+        # Yorick is very special.
+        # e.g. https://github.com/dhmunro/yorick/archive/y_2_2_04.tar.gz
+        (r'github.com/[^/]+/yorick/archive/y_(\d+(?:_\d+)*)$', path),
```

Comment:
What's so special about Yorick? It doesn't seem that fundamentally different from the Boost URL, which works fine:
```
$ spack url-parse http://downloads.sourceforge.net/project/boost/boost/1.62.0/boost_1_62_0.tar.bz2
==> Parsing URL: http://downloads.sourceforge.net/project/boost/boost/1.62.0/boost_1_62_0
    Matched regex 7: r'((\d+_)+\d+)$'

==> Detected:
    http://downloads.sourceforge.net/project/boost/boost/1.62.0/boost_1_62_0.tar.bz2
                                                                ----- ~~~~~~
    name:     boost
    version:  1_62_0

==> Substituting version 9.9.9b:
    http://downloads.sourceforge.net/project/boost/boost/1.62.0/boost_9.9.9b.tar.bz2
                                                                ----- ~~~~~~
```
I just don't like that it has "yorick" hard-coded into it's regex. It seems like it could be made more general, or made to work with an existing regex.